### PR TITLE
Remove unnecessary use cases, handle state updates at viewmodel.

### PR DESCRIPTION
**Pull request by odavidpedroza**

### CODE REVIEW

File: app/src/main/java/com/tasky/auth/network/repository/AuthRepository.kt [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
 
-    override suspend fun authenticate()  =
-        service.authenticate().isSuccessful
+    override suspend fun authenticate(): AuthResult {
+        val isAuthenticated = service.authenticate().isSuccessful
+        return if (isAuthenticated) {
+            AuthResult(Screen.Agenda)
+        } else {
+            sessionManager.clearToken()
+            AuthResult(Screen.Login)
```
Comment (philipplackner) [inline app/src/main/java/com/tasky/auth/network/repository/AuthRepository.kt:20]: It's not really the responsibility of the data layer to decide which screen to navigate to. The data layer should just provide the result, but not care what is done with it, so it also remains flexible and doesn't have to be used for this exact use case like here. In short, just return whether the user is authenticated or not and let the presentation layer decide which screen that results in :)

File: app/src/main/java/com/tasky/login/network/repository/LoginRepository.kt [hunk 2]
```diff
@@ -2,12 +2,19 @@ package com.tasky.login.network.repository
...
 ): ILoginRepository {
-    override suspend fun login(request: LoginRequest): LoginResponse? =
-        service.login(request).body()
+    override suspend fun login(email: String, password: String): Boolean {
+        val request = LoginRequest(email, password)
+        val response = service.login(request)
+        response.body()?.let {
+            sessionManager.saveAuthToken(it.token)
+        }
+        return response.body() != null
```
Comment (philipplackner) [inline app/src/main/java/com/tasky/login/network/repository/LoginRepository.kt:18]: How do you want to distinguish in the UI what kind of error happened? It's quite important to tell the user whether it was an invalid PW or missing internet connection for example. Two distinct things

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/tasky/login/network/repository/LoginRepository.kt (hunk 2 above)
Comment (odavidpedroza) [reply app/src/main/java/com/tasky/login/network/repository/LoginRepository.kt:18]: yeah, ill improve this error handling.

File: app/src/main/java/com/tasky/auth/network/repository/AuthRepository.kt (hunk 1 above)
Comment (odavidpedroza) [reply app/src/main/java/com/tasky/auth/network/repository/AuthRepository.kt:20]: that make sense, done.

**Pull request merged**